runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/ExecutionStateSampler.java

```diff
@@ -151,8 +150,10 @@ void addTracker(ExecutionStateTracker tracker) {
    *
    * <p>This method needs to be synchronized to prevent race condition with sampling thread
    */
-  synchronized void removeTracker(ExecutionStateTracker tracker) {
-    this.activeTrackers.remove(tracker);
+  void removeTracker(ExecutionStateTracker tracker) {
+    synchronized (this) {
```

Comment:
Update javadoc to match code, as needed. Or just remove commentary if not relevant.

Reply:
moved comment to member variable since that seemed more appropriate.